.github/workflows/publish.yml

```diff
@@ -0,0 +1,356 @@
+name: Publish Nargo
+
+on:
+  workflow_dispatch:
+    # Allow pushing a manual nightly release
+    inputs:
+      tag:
+        description: The tag to build Nargo from (leave empty to build a nightly release from master)
+        required: false
+      publish:
+        description: Whether to publish the build artifacts
+        type: boolean
+        default: false
+  schedule:
+    # Run a nightly release at 2 AM UTC
+    - cron: "0 2 * * *"
+  push:
```

Comment:
Should we instead make this a `merge_queue` only check?

Reply:
I think we'd want feedback on whether a release is going to fail before we actually try to run the release.

Reply:
That'd be in the merge_queue though, right? The release PR also goes through the MQ

Reply:
Such that we know every PR that lands into master will allow a release to be built, including the release PR itself.

Reply:
My point is I think we'd want that feedback before it gets to the MQ so we _know_ that the MQ will pass.

Reply:
I generally agree, but I think @kevaundray would want this check to only happen at the MQ since it takes longer than other CI stuff.

If he also wants it to run against PRs, we'd want to use `pull_request` **and** `merge_queue` events like we use on other CI stuff now.

Reply:
Ah, I get what you mean. :facepalm:

Reply:
Yeah, that seems sensible.